compiler/bytecode/instruction.go

```diff
@@ -16,55 +15,104 @@ const (
 
 // instruction actions
 const (
-	GetLocal            = "getlocal"
-	GetConstant         = "getconstant"
-	GetInstanceVariable = "getinstancevariable"
-	SetLocal            = "setlocal"
-	SetConstant         = "setconstant"
-	SetInstanceVariable = "setinstancevariable"
-	PutBoolean          = "putboolean"
-	PutString           = "putstring"
-	PutFloat            = "putfloat"
-	PutSelf             = "putself"
-	PutObject           = "putobject"
-	PutNull             = "putnil"
-	NewArray            = "newarray"
-	ExpandArray         = "expand_array"
-	SplatArray          = "splat_array"
-	NewHash             = "newhash"
-	NewRange            = "newrange"
-	BranchUnless        = "branchunless"
-	BranchIf            = "branchif"
-	Jump                = "jump"
-	Break               = "break"
-	DefMethod           = "def_method"
-	DefSingletonMethod  = "def_singleton_method"
-	DefClass            = "def_class"
-	Send                = "send"
-	InvokeBlock         = "invokeblock"
-	GetBlock            = "getblock"
-	Pop                 = "pop"
-	Dup                 = "dup"
-	Leave               = "leave"
+	GetLocal            uint8 = iota
+	GetConstant
+	GetInstanceVariable
+	SetLocal
+	SetConstant
+	SetInstanceVariable
+	PutBoolean
+	PutString
+	PutFloat
+	PutSelf
+	PutObject
+	PutNull
+	NewArray
+	ExpandArray
+	SplatArray
+	NewHash
+	NewRange
+	BranchUnless
+	BranchIf
+	Jump
+	Break
+	DefMethod
+	DefSingletonMethod
+	DefClass
+	Send
+	InvokeBlock
+	GetBlock
+	Pop
+	Dup
+	Leave
 )
 
+
+// InstructionNameTable is the table the maps instruction's op code with its readable name
+var InstructionNameTable = map[uint8]string{
```

Comment:
Prefer using [...]string{} for things like this as you are just mapping integers 0...n to strings.

Reply:
👍